tests/conftest.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import time
+import os
+import requests
+import logging.config
+import yaml
+
+def setup_logging():
+    with open('custom_logger.yaml', 'r') as config_file:
+        config = yaml.safe_load(config_file)
+        logging.config.dictConfig(config)
+
+setup_logging()
+
+logger = logging.getLogger(__name__)
+
+def pytest_html_report_title(report):
+    report.title = "CFX API Automation Report"
+
+def pytest_cmdline_preparse(args):
+    global html_report_path
+    if not os.path.exists('./reports'):
+        os.makedirs('./reports')
+    dt_str = time.strftime("%Y_%m_%d_%H_%M_%S")
+    html_file_name = "cfx_api_test_" + dt_str + ".html"
+    html_report_path = "./reports/" + html_file_name
+    print(html_report_path)
+    args.extend(['--html', html_report_path])
+    args.extend(['--self-contained-html'])
+
+@pytest.fixture(scope="session", autouse=True)
+def cleanup(request):
+    cleanup_flag = request.config.getoption("--cleanup", default=False)
+    
+    if cleanup_flag:
+        # Clear api_automation.log
+        if os.path.exists('api_automation.log'):
+            os.remove('api_automation.log')
+
+        # Clear all previous report.html files in the reports folder
+        reports_folder = 'reports'
+        if os.path.exists(reports_folder):
+            for filename in os.listdir(reports_folder):
+                if filename.startswith("cfx_api_test_") and filename.endswith(".html"):
+                    file_path = os.path.join(reports_folder, filename)
+                    os.remove(file_path)
+
+@pytest.fixture(scope="session")
+def base_url(request):
+    host = request.config.getoption("--host")
+    return f"https://{host}"
+
+@pytest.fixture(scope="session")
+def api_session(base_url, username, password):
+    # Create a session to maintain the connection
+    session = requests.Session()
+
+    try:
+        # Define test data for POST request (login)
+        post_data = {
+            "user": username,
+            "password": password
+        }
+
+        # Define headers
+        headers = {
+            'Content-Type': 'application/json'
+        }
+
+        # Perform login
+        login_url = f"{base_url}/api/v2/login"
+        response = session.post(login_url, json=post_data, headers=headers, verify=False)
+
+        if response.status_code != 200:
+            logger.error(f"Login failed with status code {response.status_code}:\n {response.text}")
+        else:
+            logger.info("::::::SESSION OPENED::::::")
+            logger.info(f"Login Success with status code {response.status_code}:\n {response.text}")
+
+        response.raise_for_status()
+
+        yield session  # Provide the session object function to test functions
+
+    finally:
+        # Close the session when all tests are done
+        session.close()
+        logger.info("::::::SESSION CLOSED::::::")
+
+def pytest_addoption(parser):
+    parser.addoption("--host", action="store", default=None, help="Platform IP address")
```

Comment:
You need to give required=True for all the mandatory options? or manually validate that all options are provided properly before continuing testing?

Reply:
these are set as mandatory siri, if we don't provide these argument in the cli it will prompt the error showing 
"pytest: error: the following arguments are required: --host", we don't have to mention required=true, as the default is None.